pubpub/platform
# Editor saving to pubfields and values

## Issue(s) Resolved
Closes https://github.com/pubpub/platform/issues/712

## High-level Explanation of PR
This PR allows for saving a pubfield of `schemaName===CoreSchemaType.String` via a rich text editor.

## Test Plan
1. Add a rich text field
2. Add the field to a form, along with another String field 
3. Fill out a form and fill out a value for the String field
4. Overwrite the field in the rich text field by typing `@[name of string field]`, hitting enter, then filling in a new value
For example:
![image](https://github.com/user-attachments/assets/a66cf7ac-0f5d-4edd-884b-c50fa29b5711)

5. Save the form
6. The form should save with the String field overwritten by what was in the Rich text field
7. Updating the form should work too, and you should be able to remove i.e. the Title field and fill in the field via the form and that should persist
8. Do the same thing on the Pub create/update form

## Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/b0b07f9f-3748-42ab-b56e-576a68887132)
![image](https://github.com/user-attachments/assets/c3ae0b95-7a3d-44f1-b9de-e41f4574553b)

https://github.com/user-attachments/assets/22d1caf9-632d-46e7-91e9-d0916f323430

## Notes
Let's say you have a pubtype with the fields Title, Content, and RichText. And you have a form that just has RichText and Title. Currently, you can specify to add Title and Content in the editor, but Content won't save! This was a "feature" of the form since we strip out values that the form doesn't have before saving the pub. But I don't know if it still makes sense with the editor.

I think it might make sense to, on the form page, limit the pubfields an editor can add to the fields that are in the form. This would require a bit of data manipulation though since the editor doesn't take pubfields directly (it takes pubtypes, then grabs those pubfields)
